### PR TITLE
Add ability for cross account import services to return streams

### LINKED ISSUE #1089: Shadow subscriptions can be leaked on stream import and connection close.
Currently if a connection is closed with open subscriptions that match a stream import the shadow subscriptions will be leaked.

### PR DESCRIPTION
**Pull request by derekcollison**
Enable stream responses. Take into account tracking of response maps that are created and do proper cleanup.

Also fixes #1089 which was discovered while working on this.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/opts.go [hunk 1]
```diff
@@ -103,18 +103,21 @@ type RemoteGatewayOpts struct {
...
+	Username          string        `json:"-"`
+	Password          string        `json:"-"`
+	AuthTimeout       float64       `json:"auth_timeout,omitempty"`
+	TLSConfig         *tls.Config   `json:"-"`
+	TLSTimeout        float64       `json:"tls_timeout,omitempty"`
+	TLSMap            bool          `json:"-"`
+	Advertise         string        `json:"-"`
+	NoAdvertise       bool          `json:"-"`
+	ReconnectInterval time.Duration `json:"-"`
+	MaxConn           int           `json:"max_connections"`
```
Comment (kozlovic) [inline server/opts.go]: Seem unrelated to this PR..

File: server/accounts.go [hunk 2]
```diff
@@ -400,6 +452,56 @@ func (a *Account) removeServiceImport(subject string) {
...
+// This tracks responses to service requests mappings. This is used for cleanup.
+func (a *Account) addRespMapEntry(acc *Account, reply, from string) {
+	a.mu.Lock()
+	if a.respMap == nil {
+		a.respMap = make(map[string][]*serviceRespEntry)
+	}
+	sre := &serviceRespEntry{acc, from}
+	sra := a.respMap[reply]
+	a.respMap[reply] = append(sra, sre)
+	if len(a.respMap) > int(a.maxnrm) {
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/accounts.go]: Good point, should probably do same trick as auto-expire ones. Will fix.

File: server/opts.go (hunk 1 above)
Comment (derekcollison) [reply server/opts.go]: Ah yes that was for leafnodes, will remove.

File: server/accounts.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/accounts.go]: ok take a look now.

File: server/accounts.go [hunk 3]
```diff
@@ -53,12 +53,14 @@ type Account struct {
 	lqws         map[string]int32
 	usersRevoked map[string]int64
 	actsRevoked  map[string]int64
+	respMap      map[string][]*serviceRespEntry
```
Comment (kozlovic) [inline server/accounts.go]: Just thinking of config reload where we do transfer things between account objects. Is that relevant here? (not necessarily the content of this map but the fact that we should move this map to the new account. I am thinking this kind of thing: https://github.com/nats-io/nats-server/blob/master/server/reload.go#L886).

Comment (derekcollison) [reply server/accounts.go]: These are runtime only, but if we reload we should transfer them over if we do not keep the live ones but make new ones.

Comment (derekcollison) [reply server/accounts.go]: ok so I think this may be an issue. It looks like we blow away all the accounts and reload them from the configuration. Note this is only for configured accounts, this does not apply to any resolver based accounts, even those from the configuration.

Comment (derekcollison) [reply server/accounts.go]: Will fix.

Comment (derekcollison) [reply server/accounts.go]: ok take another look.
